include/cache.php

```diff
@@ -84,9 +84,14 @@ public static function get($key) {
 		$memcache = self::memcache();
 		if (is_object($memcache)) {
 			// We fetch with the hostname as key to avoid problems with other applications
-			$value = $memcache->get(get_app()->get_hostname().":".$key);
-			if (!is_bool($value)) {
-				return unserialize($value);
+			$cached = $memcache->get(get_app()->get_hostname().":".$key);
+			$value = @unserialize($cached);
+
+			// Only return a value if the serialized value is valid.
+			// We also check if the db entry is a serialized
+			// boolean 'false' value (which we want to return).
+			if ($cached === 'b:0;' || $value !== false) {
+				return $value;
```

Comment:
Ugh, fair point. Then if we want a reliable way of checking the cache integrity, we should add a checksum column. The checksum of the cached value would be calculated just before serializing and would be checked against the checksum of the unserialized value. This would significantly increase the required cache storage size for small values like booleans but since serialized data takes up a lot of space already this wouldn't be too much of a problem. For example, the crc32 hash function would only adds 4 bytes and we don't really care about collisions anyway.